processor/src/main/java/org/mapstruct/ap/spi/DefaultAccessorNamingStrategy.java

```diff
@@ -166,11 +171,18 @@ public boolean isPresenceCheckMethod(ExecutableElement method) {
     @Override
     public String getPropertyName(ExecutableElement getterOrSetterMethod) {
         String methodName = getterOrSetterMethod.getSimpleName().toString();
-        if ( methodName.startsWith( "get" ) || methodName.startsWith( "set" ) ) {
-            return IntrospectorUtils.decapitalize( methodName.substring( 3 ) );
-        }
-        else if ( isFluentSetter( getterOrSetterMethod ) ) {
-            return methodName;
+        if ( isFluentSetter( getterOrSetterMethod ) ) {
+            // If this is a fluent setter that starts with get or set and the 4th character is an uppercase one
+            // then we treat it as a Java Bean style method (we get the property starting from the 4th character).
+            // Otherwise we treat it as a fluent setter
+            if ( ( methodName.startsWith( "get" ) || methodName.startsWith( "set" ) )
+                && methodName.length() > 3
+                && Character.isUpperCase( methodName.charAt( 3 ) ) ) {
```

Comment:
I remember this was topic of discussion right? I mean, whether the third letter must be capital or not?

Reply:
Yes it was a topic and to me the most sense was to treat strictly uppercase 4th character for fluent setter. This are special kind of setters that return the same instance back.

For example:

```
public Builder setSettlementDate(String settlementDate);
public Builder settlementDate(String settlementDate);
```

If someone uses something else for fluent setters then we don't support that.

Reply:
would be good to add this particular example to the comment.

Reply:
Updated. Does it make more sense now?

Reply:
yup..